dali/kernels/slice/slice_gpu.cuh

```diff
@@ -46,6 +46,30 @@ struct BlockDesc {
   std::size_t size;
 };
 
+template <int Dims, typename OutputType, typename InputType>
+__device__ void SliceFunc(OutputType *__restrict__ out, const InputType *__restrict__ in,
+    const int64_t *out_strides, const int64_t *in_strides,
+    std::size_t offset, std::size_t block_end) {
+  if (Dims > 1 && out_strides[Dims-1] == in_strides[Dims-1]) {
+    const int NextDims = Dims > 1 ? Dims-1 : 1;
+    SliceFunc<NextDims>(out, in, out_strides, in_strides, offset, block_end);
+    return;
+  }
+
+  for (; offset < block_end; offset += blockDim.x) {
+    int64_t idx = offset;
```

Comment:
didn't we agree to use std::size_t for indexes?

Reply:
It seems that using unsigned types was a bit slower, at least on my machine. Anyway, we use int64_t in SampleDesc.